Index: src/main/resources/com/brunycharotte/mastermindfx/MasterMind_FX.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/com/brunycharotte/mastermindfx/MasterMind_FX.fxml b/src/main/resources/com/brunycharotte/mastermindfx/MasterMindPlayer.fxml
rename from src/main/resources/com/brunycharotte/mastermindfx/MasterMind_FX.fxml
rename to src/main/resources/com/brunycharotte/mastermindfx/MasterMindPlayer.fxml
--- a/src/main/resources/com/brunycharotte/mastermindfx/MasterMind_FX.fxml	
+++ b/src/main/resources/com/brunycharotte/mastermindfx/MasterMindPlayer.fxml	
@@ -1,25 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<?import javafx.geometry.Insets?>
-<?import javafx.scene.control.Button?>
-<?import javafx.scene.control.Label?>
-<?import javafx.scene.image.Image?>
-<?import javafx.scene.image.ImageView?>
-<?import javafx.scene.layout.ColumnConstraints?>
-<?import javafx.scene.layout.GridPane?>
-<?import javafx.scene.layout.HBox?>
-<?import javafx.scene.layout.Pane?>
-<?import javafx.scene.layout.RowConstraints?>
-<?import javafx.scene.layout.VBox?>
-<?import javafx.scene.shape.Circle?>
-<?import javafx.scene.shape.Line?>
-<?import javafx.scene.text.Font?>
+<?import javafx.geometry.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.scene.shape.*?>
+<?import javafx.scene.text.*?>
 
-<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/19" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.MainControler">
+<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="853.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/17.0.2-ea" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.MainControler">
    <children>
-      <Pane id="scene" fx:id="logoJoueurCourant" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" styleClass="paneScene" stylesheets="@style.css">
+      <Pane id="scene" fx:id="logoJoueurCourant" layoutX="-3.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="862.0" styleClass="paneScene" stylesheets="@style.css">
          <children>
-            <VBox id="plateauVbox" fx:id="plateau" layoutX="287.0" layoutY="78.0" prefHeight="483.0" prefWidth="352.0" spacing="4.0" style="-fx-background-radius: 30; -fx-border-radius: 30; -fx-effect: dropshadow(three-pass-box, rgba(0,0,0,0.8), 10, 0, 0, 0);">
+            <VBox id="plateauVbox" fx:id="plateau" layoutX="274.0" layoutY="78.0" prefHeight="483.0" prefWidth="352.0" spacing="4.0" style="-fx-background-radius: 30; -fx-border-radius: 30; -fx-effect: dropshadow(three-pass-box, rgba(0,0,0,0.8), 10, 0, 0, 0);">
                <children>
                   <HBox id="coup9" alignment="CENTER" prefHeight="31.0" prefWidth="123.0" spacing="9.0" style="-fx-background-color: transparent;">
                      <VBox.margin>
@@ -403,47 +395,47 @@
                   <Insets bottom="20.0" left="20.0" right="20.0" top="20.0" />
                </padding>
             </VBox>
-            <Button id="boutonQuitter" fx:id="boutonQuitter" alignment="CENTER" layoutX="681.0" layoutY="499.0" mnemonicParsing="false" onAction="#afficherQuitterMenu" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Quitter la partie">
+            <Button id="boutonQuitter" fx:id="boutonQuitter" alignment="CENTER" layoutX="25.0" layoutY="510.0" mnemonicParsing="false" onAction="#afficherQuitterMenu" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Quitter la partie">
                <font>
                   <Font name="Agency FB Bold" size="30.0" />
                </font>
             </Button>
-            <Button id="boutonEffacer" fx:id="boutonEffacer" layoutX="681.0" layoutY="367.0" mnemonicParsing="false" onAction="#removeLastCircle" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Effacer">
+            <Button id="boutonEffacer" fx:id="boutonEffacer" layoutX="642.0" layoutY="433.0" mnemonicParsing="false" onAction="#removeLastCircle" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Effacer">
                <font>
                   <Font name="Agency FB" size="12.0" />
                </font>
             </Button>
-            <Button id="boutonValider" fx:id="validerButton" layoutX="681.0" layoutY="433.0" mnemonicParsing="false" onAction="#envoyerCode" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Valider">
+            <Button id="boutonValider" fx:id="validerButton" layoutX="642.0" layoutY="510.0" mnemonicParsing="false" onAction="#envoyerCode" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Valider">
                <font>
                   <Font name="Agency FB" size="12.0" />
                </font>
             </Button>
-            <Pane id="palette" fx:id="palette1" layoutX="681.0" layoutY="78.0" prefHeight="270.0" prefWidth="192.0" styleClass="palette">
+            <Pane id="palette" fx:id="palette1" layoutX="642.0" layoutY="90.0" prefHeight="326.0" prefWidth="192.0" styleClass="palette">
                <children>
-                  <Label alignment="CENTER" layoutX="16.0" prefHeight="50.0" prefWidth="150.0" styleClass="labelDroite" stylesheets="@style.css" text="Palette de couleur" textAlignment="CENTER" wrapText="true">
+                  <Label alignment="CENTER" layoutX="21.0" layoutY="14.0" prefHeight="50.0" prefWidth="150.0" styleClass="labelDroite" stylesheets="@style.css" text="Palette de couleur" textAlignment="CENTER" wrapText="true">
                      <font>
                         <Font name="Agency FB" size="24.0" />
                      </font>
                   </Label>
-                  <Button fx:id="rouge" layoutX="112.0" layoutY="50.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickRed" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: red;" />
-                  <Button fx:id="yellow" layoutX="112.0" layoutY="194.0" maxHeight="-Infinity" maxWidth="1.7976931348623157E308" minHeight="0.0" minWidth="53.0" mnemonicParsing="false" onAction="#onClickJaune" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: yellow;" />
-                  <Button fx:id="vert" layoutX="21.0" layoutY="50.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickVert" prefHeight="51.0" prefWidth="30.0" style="-fx-background-radius: 30; -fx-background-color: green;" />
-                  <Button fx:id="bleu" layoutX="21.0" layoutY="118.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickBlue" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: blue;" />
-                  <Button fx:id="orange" layoutX="21.0" layoutY="194.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickOrange" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: orange;" />
-                  <Button fx:id="violet" layoutX="112.0" layoutY="118.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickViolet" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: purple;" />
+                  <Button fx:id="rouge" layoutX="115.0" layoutY="80.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickRed" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: red;" />
+                  <Button fx:id="yellow" layoutX="115.0" layoutY="239.0" maxHeight="-Infinity" maxWidth="1.7976931348623157E308" minHeight="0.0" minWidth="53.0" mnemonicParsing="false" onAction="#onClickJaune" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: yellow;" />
+                  <Button fx:id="vert" layoutX="24.0" layoutY="80.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickVert" prefHeight="51.0" prefWidth="30.0" style="-fx-background-radius: 30; -fx-background-color: green;" />
+                  <Button fx:id="bleu" layoutX="24.0" layoutY="158.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickBlue" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: blue;" />
+                  <Button fx:id="orange" layoutX="24.0" layoutY="239.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickOrange" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: orange;" />
+                  <Button fx:id="violet" layoutX="115.0" layoutY="158.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="53.0" mnemonicParsing="false" onAction="#onClickViolet" prefHeight="51.0" prefWidth="53.0" style="-fx-background-radius: 30; -fx-background-color: purple;" />
                </children>
             </Pane>
-            <Button fx:id="changerButton" layoutX="681.0" layoutY="433.0" mnemonicParsing="false" onAction="#switchToBot" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Prochaine Manche" visible="false">
+            <Button fx:id="changerButton" layoutX="673.0" layoutY="461.0" mnemonicParsing="false" onAction="#switchToBot" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Prochaine Manche" visible="false">
                <font>
                   <Font name="Agency FB" size="12.0" />
                </font>
             </Button>
-            <Label id="instruction" alignment="CENTER" layoutX="345.0" layoutY="14.0" prefHeight="50.0" prefWidth="237.0" styleClass="labelTitle" text="MASTERMIND">
+            <Label id="instruction" alignment="CENTER" layoutX="332.0" layoutY="14.0" prefHeight="50.0" prefWidth="237.0" styleClass="labelTitle" text="MASTERMIND">
                <font>
                   <Font name="Agency FB" size="36.0" />
                </font>
             </Label>
-            <Label fx:id="secretCodeLabel" layoutX="681.0" layoutY="369.0" prefHeight="57.0" prefWidth="192.0" styleClass="labelGauche" visible="false">
+            <Label fx:id="secretCodeLabel" layoutX="673.0" layoutY="397.0" prefHeight="57.0" prefWidth="192.0" styleClass="labelGauche" visible="false">
                <graphic>
                   <HBox fx:id="rowSecretCode" alignment="CENTER" prefHeight="65.0" prefWidth="161.0" spacing="10.0">
                      <children>
@@ -455,12 +447,7 @@
                   </HBox>
                </graphic>
             </Label>
-            <Label id="instruction" fx:id="labelManche" alignment="CENTER" layoutX="24.0" layoutY="78.0" prefHeight="85.0" prefWidth="237.0" styleClass="labelGauche" text="Manche 1">
-               <font>
-                  <Font name="Agency FB" size="36.0" />
-               </font>
-            </Label>
-            <GridPane alignment="CENTER" layoutX="24.0" layoutY="314.0" prefHeight="137.0" prefWidth="237.0" styleClass="labelScore">
+            <GridPane alignment="CENTER" layoutX="25.0" layoutY="279.0" prefHeight="137.0" prefWidth="237.0" styleClass="labelScore">
                <columnConstraints>
                   <ColumnConstraints hgrow="SOMETIMES" maxWidth="113.20001220703125" minWidth="10.0" prefWidth="101.799951171875" />
                   <ColumnConstraints hgrow="SOMETIMES" maxWidth="75.79998779296875" minWidth="0.0" prefWidth="27.400048828124994" />
@@ -511,17 +498,22 @@
                   </Label>
                </children>
             </GridPane>
-            <Button id="boutonHistorique" fx:id="boutonHistorique" alignment="CENTER" layoutX="24.0" layoutY="478.0" mnemonicParsing="false" onAction="#afficherHistorique" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Historique des manches">
+            <Button id="boutonHistorique" fx:id="boutonHistorique" alignment="CENTER" layoutX="25.0" layoutY="433.0" mnemonicParsing="false" onAction="#afficherHistorique" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Historique des manches">
                <font>
                   <Font name="Agency FB Bold" size="30.0" />
                </font>
             </Button>
-            <Label id="instruction" alignment="CENTER" layoutX="24.0" layoutY="201.0" prefHeight="85.0" prefWidth="237.0" styleClass="labelGauche" text="A toi de décoder !">
+            <Label id="instruction" alignment="CENTER" layoutX="47.0" layoutY="192.0" prefHeight="71.0" prefWidth="192.0" styleClass="labelGauche" text="A toi de décoder !">
+               <font>
+                  <Font name="Agency FB" size="36.0" />
+               </font>
+            </Label>
+            <Label id="instruction" fx:id="labelManche" alignment="CENTER" layoutX="47.0" layoutY="97.0" prefHeight="85.0" prefWidth="192.0" styleClass="labelGauche" text="Manche 1">
                <font>
                   <Font name="Agency FB" size="36.0" />
                </font>
             </Label>
-            <ImageView fitHeight="50.0" fitWidth="50.0" layoutX="118.0" layoutY="39.0" pickOnBounds="true" preserveRatio="true">
+            <ImageView fitHeight="50.0" fitWidth="50.0" layoutX="118.0" layoutY="65.0" pickOnBounds="true" preserveRatio="true">
                <image>
                   <Image url="@icone_joueur.png" />
                </image>
Index: src/main/resources/com/brunycharotte/mastermindfx/MasterMindBot.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/com/brunycharotte/mastermindfx/MasterMindBot.fxml b/src/main/resources/com/brunycharotte/mastermindfx/MasterMindBot.fxml
--- a/src/main/resources/com/brunycharotte/mastermindfx/MasterMindBot.fxml	
+++ b/src/main/resources/com/brunycharotte/mastermindfx/MasterMindBot.fxml	
@@ -1,26 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<?import javafx.geometry.Insets?>
-<?import javafx.scene.control.Button?>
-<?import javafx.scene.control.Label?>
-<?import javafx.scene.control.Slider?>
-<?import javafx.scene.image.Image?>
-<?import javafx.scene.image.ImageView?>
-<?import javafx.scene.layout.ColumnConstraints?>
-<?import javafx.scene.layout.GridPane?>
-<?import javafx.scene.layout.HBox?>
-<?import javafx.scene.layout.Pane?>
-<?import javafx.scene.layout.RowConstraints?>
-<?import javafx.scene.layout.VBox?>
-<?import javafx.scene.shape.Circle?>
-<?import javafx.scene.shape.Line?>
-<?import javafx.scene.text.Font?>
+<?import javafx.geometry.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.scene.shape.*?>
+<?import javafx.scene.text.*?>
 
-<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/19" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.BotControler">
+<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="862.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/17.0.2-ea" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.BotControler">
    <children>
-      <Pane id="scene" fx:id="gamePanel" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css">
+      <Pane id="scene" fx:id="gamePanel" layoutX="6.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="862.0" stylesheets="@style.css">
          <children>
-            <Pane id="palette" fx:id="palette" layoutX="681.0" layoutY="82.0" prefHeight="264.0" prefWidth="192.0" styleClass="palette">
+            <Pane id="palette" fx:id="palette" layoutX="642.0" layoutY="90.0" prefHeight="264.0" prefWidth="192.0" styleClass="palette">
                <children>
                   <Label alignment="CENTER" layoutX="7.0" layoutY="14.0" prefHeight="65.0" prefWidth="179.0" styleClass="labelDroite" stylesheets="@style.css" text="Combien de jetons bien placés ?" textAlignment="CENTER" wrapText="true">
                      <font>
@@ -36,7 +27,7 @@
                   <Slider fx:id="malPlaceSlider" blockIncrement="1.0" layoutX="15.0" layoutY="201.0" majorTickUnit="1.0" max="4.0" minorTickCount="0" prefHeight="42.0" prefWidth="163.0" showTickLabels="true" showTickMarks="true" snapToTicks="true" styleClass="sliderMalPlaces" />
                </children>
             </Pane>
-            <VBox id="plateauVbox" fx:id="plateau" layoutX="287.0" layoutY="78.0" prefHeight="483.0" prefWidth="352.0" spacing="4.0" style="-fx-background-radius: 30; -fx-border-radius: 30; -fx-effect: dropshadow(three-pass-box, rgba(0,0,0,0.8), 10, 0, 0, 0);">
+            <VBox id="plateauVbox" fx:id="plateau" layoutX="274.0" layoutY="78.0" prefHeight="483.0" prefWidth="352.0" spacing="4.0" style="-fx-background-radius: 30; -fx-border-radius: 30; -fx-effect: dropshadow(three-pass-box, rgba(0,0,0,0.8), 10, 0, 0, 0);">
                <children>
                   <HBox id="coup9" alignment="CENTER" prefHeight="31.0" prefWidth="123.0" spacing="9.0" style="-fx-background-color: transparent;">
                      <VBox.margin>
@@ -420,17 +411,17 @@
                   <Insets bottom="20.0" left="20.0" right="20.0" top="20.0" />
                </padding>
             </VBox>
-            <Button id="boutonValider" fx:id="validerButton" layoutX="681.0" layoutY="433.0" mnemonicParsing="false" onAction="#updateBienMalPlace" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Valider">
+            <Button id="boutonValider" fx:id="validerButton" layoutX="642.0" layoutY="433.0" mnemonicParsing="false" onAction="#updateBienMalPlace" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Valider">
                <font>
                   <Font name="Agency FB" size="12.0" />
                </font>
             </Button>
-            <Button id="boutonQuitter" fx:id="boutonQuitter" alignment="CENTER" layoutX="681.0" layoutY="499.0" mnemonicParsing="false" onAction="#afficherQuitterMenu" prefHeight="57.0" prefWidth="192.0" styleClass="bouton" stylesheets="@style.css" text="Quitter la partie">
+            <Button id="boutonQuitter" fx:id="boutonQuitter" alignment="CENTER" layoutX="25.0" layoutY="510.0" mnemonicParsing="false" onAction="#afficherQuitterMenu" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Quitter la partie">
                <font>
                   <Font name="Agency FB Bold" size="30.0" />
                </font>
             </Button>
-            <Label id="instruction" alignment="CENTER" layoutX="345.0" layoutY="14.0" prefHeight="50.0" prefWidth="237.0" styleClass="labelTitle" text="MASTERMIND">
+            <Label id="instruction" alignment="CENTER" layoutX="332.0" layoutY="14.0" prefHeight="50.0" prefWidth="237.0" styleClass="labelTitle" text="MASTERMIND">
                <font>
                   <Font name="Agency FB" size="36.0" />
                </font>
@@ -440,24 +431,13 @@
                   <Font name="Agency FB" size="12.0" />
                </font>
             </Button>
-            <Label layoutX="681.0" layoutY="359.0" prefHeight="57.0" prefWidth="192.0" styleClass="labelGauche">
-               <graphic>
-                  <HBox fx:id="rowSecretCode" alignment="CENTER" prefHeight="65.0" prefWidth="161.0" spacing="10.0">
-                     <children>
-                        <Circle fx:id="circle1" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
-                        <Circle fx:id="circle2" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
-                        <Circle fx:id="circle3" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
-                        <Circle fx:id="circle4" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
-                     </children>
-                  </HBox>
-               </graphic>
-            </Label>
-            <Label id="instruction" fx:id="labelManche" alignment="CENTER" layoutX="24.0" layoutY="78.0" prefHeight="85.0" prefWidth="237.0" styleClass="labelGauche" text="Manche 1">
+            <Label layoutX="642.0" layoutY="364.0" prefHeight="57.0" prefWidth="192.0" styleClass="labelGauche" />
+            <Label id="instruction" fx:id="labelManche" alignment="CENTER" layoutX="47.0" layoutY="97.0" prefHeight="85.0" prefWidth="192.0" styleClass="labelGauche" text="Manche 1">
                <font>
                   <Font name="Agency FB" size="36.0" />
                </font>
             </Label>
-            <GridPane alignment="CENTER" layoutX="24.0" layoutY="314.0" prefHeight="137.0" prefWidth="237.0" styleClass="labelScore">
+            <GridPane alignment="CENTER" layoutX="25.0" layoutY="279.0" prefHeight="137.0" prefWidth="237.0" styleClass="labelScore">
                <columnConstraints>
                   <ColumnConstraints hgrow="SOMETIMES" maxWidth="113.20001220703125" minWidth="10.0" prefWidth="101.799951171875" />
                   <ColumnConstraints hgrow="SOMETIMES" maxWidth="75.79998779296875" minWidth="0.0" prefWidth="27.400048828124994" />
@@ -508,21 +488,29 @@
                   </Label>
                </children>
             </GridPane>
-            <Button id="boutonHistorique" fx:id="boutonHistorique" alignment="CENTER" layoutX="24.0" layoutY="478.0" mnemonicParsing="false" onAction="#afficherHistorique" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Historique des manches">
+            <Button id="boutonHistorique" fx:id="boutonHistorique" alignment="CENTER" layoutX="25.0" layoutY="433.0" mnemonicParsing="false" onAction="#afficherHistorique" prefHeight="57.0" prefWidth="237.0" styleClass="bouton" stylesheets="@style.css" text="Historique des manches">
                <font>
                   <Font name="Agency FB Bold" size="30.0" />
                </font>
             </Button>
-            <Label id="instruction" alignment="CENTER" layoutX="24.0" layoutY="201.0" prefHeight="85.0" prefWidth="237.0" styleClass="labelGauche" text="A toi de décoder !">
+            <Label id="instruction" alignment="CENTER" layoutX="47.0" layoutY="192.0" prefHeight="71.0" prefWidth="192.0" styleClass="labelGauche" text="A toi de coder !">
                <font>
                   <Font name="Agency FB" size="36.0" />
                </font>
             </Label>
-            <ImageView fitHeight="50.0" fitWidth="50.0" layoutX="118.0" layoutY="39.0" pickOnBounds="true" preserveRatio="true">
+            <ImageView fitHeight="50.0" fitWidth="50.0" layoutX="118.0" layoutY="65.0" pickOnBounds="true" preserveRatio="true">
                <image>
                   <Image url="@icone_bot.png" />
                </image>
             </ImageView>
+            <HBox fx:id="rowSecretCode" alignment="CENTER" layoutX="658.0" layoutY="360.0" prefHeight="65.0" prefWidth="161.0" spacing="10.0">
+               <children>
+                  <Circle fx:id="circle1" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
+                  <Circle fx:id="circle2" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
+                  <Circle fx:id="circle3" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
+                  <Circle fx:id="circle4" fill="WHITE" radius="15.0" stroke="BLACK" strokeType="INSIDE" strokeWidth="0.0" />
+               </children>
+            </HBox>
          </children>
       </Pane>
       <Pane fx:id="historiquePane1" onMouseClicked="#enleverHistorique" prefHeight="588.0" prefWidth="900.0" style="-fx-background-color: transparent;" stylesheets="@style.css" visible="false">
Index: src/main/java/com/brunycharotte/mastermindfx/TutoView.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/brunycharotte/mastermindfx/TutoView.java b/src/main/java/com/brunycharotte/mastermindfx/TutoView.java
--- a/src/main/java/com/brunycharotte/mastermindfx/TutoView.java	
+++ b/src/main/java/com/brunycharotte/mastermindfx/TutoView.java	
@@ -40,7 +40,7 @@
             displayedImage.setImage(images[indexImage]);
         }
         else {
-            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMind_FX.fxml"));
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMindPlayer.fxml"));
             root = fxmlLoader.load();
             MainControler controler = fxmlLoader.getController();
             stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
@@ -62,7 +62,7 @@
     }
 
     public void onPlay(ActionEvent event) throws IOException {
-        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMind_FX.fxml"));
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMindPlayer.fxml"));
         root = fxmlLoader.load();
         MainControler controler = fxmlLoader.getController();
         stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
Index: src/main/resources/com/brunycharotte/mastermindfx/EndingView.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/com/brunycharotte/mastermindfx/EndingView.fxml b/src/main/resources/com/brunycharotte/mastermindfx/EndingView.fxml
--- a/src/main/resources/com/brunycharotte/mastermindfx/EndingView.fxml	
+++ b/src/main/resources/com/brunycharotte/mastermindfx/EndingView.fxml	
@@ -1,22 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<?import javafx.geometry.Insets?>
-<?import javafx.scene.control.Button?>
-<?import javafx.scene.control.Label?>
-<?import javafx.scene.image.Image?>
-<?import javafx.scene.image.ImageView?>
-<?import javafx.scene.layout.ColumnConstraints?>
-<?import javafx.scene.layout.GridPane?>
-<?import javafx.scene.layout.Pane?>
-<?import javafx.scene.layout.RowConstraints?>
-<?import javafx.scene.layout.VBox?>
-<?import javafx.scene.text.Font?>
+<?import javafx.geometry.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.scene.text.*?>
 
-<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/19" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.EndingControler">
+<Pane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css" xmlns="http://javafx.com/javafx/17.0.2-ea" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.brunycharotte.mastermindfx.EndingControler">
    <children>
       <Pane id="scene" fx:id="gamePanel" layoutX="10.0" layoutY="10.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="588.0" prefWidth="900.0" stylesheets="@style.css">
          <children>
-            <ImageView id="logo" fitHeight="192.0" fitWidth="192.0" layoutX="47.0" layoutY="14.0" pickOnBounds="true" preserveRatio="true">
+            <ImageView id="logo" fitHeight="192.0" fitWidth="192.0" layoutX="14.0" layoutY="14.0" pickOnBounds="true" preserveRatio="true">
                <image>
                   <Image url="@logo.png" />
                </image>
Index: src/main/java/com/brunycharotte/mastermindfx/EndingControler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/brunycharotte/mastermindfx/EndingControler.java b/src/main/java/com/brunycharotte/mastermindfx/EndingControler.java
--- a/src/main/java/com/brunycharotte/mastermindfx/EndingControler.java	
+++ b/src/main/java/com/brunycharotte/mastermindfx/EndingControler.java	
@@ -31,7 +31,7 @@
     Stage stage;
 
     public void restart(ActionEvent event) throws IOException {
-        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMind_FX.fxml"));
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMindPlayer.fxml"));
         root = fxmlLoader.load();
         stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
         scene = new Scene(root);
Index: src/main/java/com/brunycharotte/mastermindfx/BotControler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/brunycharotte/mastermindfx/BotControler.java b/src/main/java/com/brunycharotte/mastermindfx/BotControler.java
--- a/src/main/java/com/brunycharotte/mastermindfx/BotControler.java	
+++ b/src/main/java/com/brunycharotte/mastermindfx/BotControler.java	
@@ -134,7 +134,7 @@
 
     public void switchToHuman(ActionEvent event) throws IOException {
         if (manche != 10) {
-            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMind_FX.fxml"));
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("MasterMindPlayer.fxml"));
             root = fxmlLoader.load();
             MainControler controler = fxmlLoader.getController();
             stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
